Packet++/src/LdapLayer.cpp

```diff
@@ -0,0 +1,210 @@
+#include "LdapLayer.h"
+#include "GeneralUtils.h"
+#include <unordered_map>
+
+namespace pcpp {
+
+	// region LdapOperationType
+
+	const std::unordered_map<LdapOperationType::Value, std::string, EnumClassHash<LdapOperationType::Value>> LdapOperationTypeToString{
+		{LdapOperationType::BindRequest,           "BindRequest"},
+		{LdapOperationType::BindResponse,          "BindResponse"},
+		{LdapOperationType::UnbindRequest,         "UnbindRequest"},
+		{LdapOperationType::SearchRequest,         "SearchRequest"},
+		{LdapOperationType::SearchResultEntry,     "SearchResultEntry"},
+		{LdapOperationType::SearchResultDone,      "SearchResultDone"},
+		{LdapOperationType::ModifyRequest,         "ModifyRequest"},
+		{LdapOperationType::ModifyResponse,        "ModifyResponse"},
+		{LdapOperationType::AddRequest,            "AddRequest"},
+		{LdapOperationType::AddResponse,           "AddResponse"},
+		{LdapOperationType::DelRequest,            "DelRequest"},
+		{LdapOperationType::DelResponse,           "DelResponse"},
+		{LdapOperationType::ModifyDNRequest,       "ModifyDNRequest"},
+		{LdapOperationType::ModifyDNResponse,      "ModifyDNResponse"},
+		{LdapOperationType::CompareRequest,        "CompareRequest"},
+		{LdapOperationType::CompareResponse,       "CompareResponse"},
+		{LdapOperationType::AbandonRequest,        "AbandonRequest"},
+		{LdapOperationType::SearchResultReference, "SearchResultReference"},
+		{LdapOperationType::ExtendedRequest,       "ExtendedRequest"},
+		{LdapOperationType::ExtendedResponse,      "ExtendedResponse"},
+		{LdapOperationType::IntermediateResponse,  "IntermediateResponse"},
+		{LdapOperationType::Unknown,               "Unknown"}
+	};
+
+	const std::unordered_map<uint8_t, LdapOperationType> UintToLdapOperationType{
```

Comment:
nit: Maybe add a sanity check `static_assert` that `LdapOperationType::Value`'s backing field is not larger than `uint8_t`, since we are initialzing via `static_cast`s?

Reply:
`LdapOperationType::Value` is an enum of type `uint8_t`: https://github.com/seladb/PcapPlusPlus/blob/394de3ebac90d6a91674e7ecbe7f9e888129b3c7/Packet%2B%2B/header/LdapLayer.h#L27

So I think it's safe to assume all values are of type `uint8_t`. Besides, this is an internal map that doesn't rely on any user input so I'm not sure this check is necessary

Reply:
if it's only internal, probably the current implemetation is fine.